api/controllers/kubernetes/install/controller.go

```diff
@@ -195,24 +194,25 @@ func NewInstallController(opts ...InstallControllerOption) (*InstallController,
 		controller.infraManager = infraManager
 	}
 
-	if controller.appConfigManager == nil {
-		controller.appConfigManager = appconfig.NewAppConfigManager(
-			appconfig.WithLogger(controller.logger),
-			appconfig.WithAppConfigStore(controller.store.AppConfigStore()),
-		)
-	}
-
-	if controller.configValues != nil {
-		if controller.releaseData == nil || controller.releaseData.AppConfig == nil {
-			return nil, errors.New("app config not found")
+	if controller.releaseData != nil && controller.releaseData.AppConfig != nil {
+		// Initialize the app config manager if an app config is provided
+		if controller.appConfigManager == nil {
```

Comment:
shouldnt we always instantiate this and instead fail on nil controller.releaseData.AppConfig?